controllers/mock.go

```diff
@@ -28,13 +28,13 @@ func (i ModelInterfaceMock) NewProduct(name string, public bool, detailsID *uuid
 	return &p, i.err
```

Comment:
In these, situations you could create pointer objects and return them instead of creating objects and create pointers from them:
```
p := &models.Product{...}
return p, i.err
```